backend/amd64/emit.mlp

```diff
@@ -988,6 +988,59 @@ let emit_atomic instr op (size : Cmm.atomic_bitwidth) addr =
     I.set E res8;
     I.movzx res8 res
 
+let emit_reinterpret_cast (cast : Cmm.reinterpret_cast) i =
+  let distinct = i.arg.(0).loc <> i.res.(0).loc in
```

Comment:
(Not part of this pull request, but we
should probably avoid the polymorphic
comparison - same in `emit_static_cast`.)